plugin/parser/engine/pg/parser.go

```diff
@@ -28,13 +30,22 @@ func (p *PostgreSQLParser) Parse(ctx parser.Context, statement string) ([]ast.No
 		return nil, err
 	}
 
+	textList, err := parser.SplitMultiSQL(parser.Postgres, statement)
+	if err != nil {
+		return nil, err
+	}
+	if len(res.Stmts) != len(textList) {
+		return nil, fmt.Errorf("split multi-SQL failed: the length should be %d, but get %d. stmt: \"%s\"", len(res.Stmts), len(textList), statement)
+	}
+
 	var nodeList []ast.Node
 
-	for _, stmt := range res.Stmts {
+	for i, stmt := range res.Stmts {
 		node, err := convert(stmt.Stmt)
 		if err != nil {
 			return nil, err
 		}
+		node.SetText(textList[i])
```

Comment:
What prevents us call convert(textList[i]) in the first place and avoid calling node.SetText here?

Reply:
Sorry, what does "the first place" means?